GCRCatalogs/register.py

```diff
@@ -142,6 +159,17 @@ def root_dir(self, path):
             warnings.warn("root_dir has been set to '{}' but errors may occur when you try to access it: {}".format(path, e))
         self._custom_root_dir = path
 
+    def set_root_dir_by_site(self, site):
```

Comment:
Make `site_list` an on-the-fly property
```suggestion
    @property
    def site_list(self):
        return list(self._site_config)

    def set_root_dir_by_site(self, site):
```